### PR TITLE
(4) Move transaction/span related functions from Hub to Scope

### PR DESCRIPTION
**Pull request by antonpirker**
Moved some functionality from Hub to Client:

- moved `start_transaction` from Hub to Scope
- moved `start_span` from Hub to Scope
- moved `continue_trace` from Hub to Scope

This is preparation work for refactoring how we deal with Hubs and Scopes in the future.

Depends on: https://github.com/getsentry/sentry-python/pull/2558 (please review the linked one first)

### CODE REVIEW

**Review by sentrivana — APPROVED** [review]

File: sentry_sdk/scope.py [hunk 1]
```diff
@@ -636,6 +637,155 @@ def add_breadcrumb(self, crumb=None, hint=None, **kwargs):
...
+        configuration_instrumenter = client and client.options["instrumenter"]
+
+        if instrumenter != configuration_instrumenter:
+            return NoOpSpan()
+
+        custom_sampling_context = kwargs.pop("custom_sampling_context", {})
+
+        # if we haven't been given a transaction, make one
+        if transaction is None:
+            kwargs.setdefault("hub", hub)
```
Comment (sl0thentr0py) [inline sentry_sdk/scope.py:680]: so this shows that these Transaction functions have a dependency on hub. We will need to think about how to get rid of that.

The only non-trivial usage is in the context manager 
https://github.com/getsentry/sentry-python/blob/ddf37a335c16e0b8e07c5904cc49011aea7264dd/sentry_sdk/tracing.py#L197-L216

File: sentry_sdk/scope.py [hunk 2]
```diff
@@ -636,6 +637,155 @@ def add_breadcrumb(self, crumb=None, hint=None, **kwargs):
...
+
+        if instrumenter != configuration_instrumenter:
+            return NoOpSpan()
+
+        # THIS BLOCK IS DEPRECATED
+        # TODO: consider removing this in a future release.
+        # This is for backwards compatibility with releases before
+        # start_transaction existed, to allow for a smoother transition.
+        if isinstance(span, Transaction) or "transaction" in kwargs:
+            deprecation_msg = (
```
Comment (sl0thentr0py) [inline sentry_sdk/scope.py:734]: let us make sure to remove these in the major. In general, we should massively simplify this span behaviour - we don't need to have 5 different ways of using this function. One well documented way is sufficient.

**Review by sl0thentr0py — APPROVED** [review]
lgtm, just some observations

File: sentry_sdk/scope.py (hunk 1 above)
Comment (antonpirker) [reply sentry_sdk/scope.py:680]: True. Will change that in a separate pr.

File: sentry_sdk/scope.py (hunk 2 above)
Comment (antonpirker) [reply sentry_sdk/scope.py:734]: Yes, it is part of this PR: https://github.com/getsentry/sentry-python/pull/2581

**Pull request merged**